lib/services/access_checker.go

```diff
@@ -1060,7 +1060,7 @@ func (a *accessChecker) HostUsers(s types.Server) (*HostUsersInfo, error) {
 
 		hostUserShell := role.GetOptions().CreateHostUserDefaultShell
 		if hostUserShell != "" {
-			if shell != "" {
+			if shell == "" {
```

Comment:
This was definitely working at one point, so I must've fat fingered this comparison :facepalm: Added a test to catch this in the future

Reply:
Can this whole thing collapse down to a one liner? `shell = cmp.Or(shell, role.GetOptions().CreateHostUserDefaultShell)`?

- If `shell` is already set it's a no-op.
- If `shell` is not set and the role doesn't specify a default `shell` remains empty
- If `shell` is not set and the role does specify a shell, then `shell` is overwritten

Reply:
Good call, shell assignment totally can be :+1: I really need to start reaching for the `cmp` package more often